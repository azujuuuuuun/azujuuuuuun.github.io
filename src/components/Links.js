import React from 'react'
import { Link } from 'gatsby'

export default props => {
  const { listStyle, isFooter, itemStyle, linkStyle } = props
  return (
    <ul style={listStyle}>
      {isFooter && (
        <li style={itemStyle}>
          <Link to="/" style={linkStyle}>ホーム</Link>
        </li>
      )}
      <li style={itemStyle}>
        <Link to="/about" style={linkStyle}>自己紹介</Link>
      </li>
      <li style={itemStyle}>
        <Link to="/skills" style={linkStyle}>スキル</Link>
      </li>
      <li style={itemStyle}>
        <Link to="/favorites" style={linkStyle}>好きなもの</Link>
      </li>
      <li style={itemStyle}>
        <Link to="/links" style={linkStyle}>リンク</Link>
      </li>
    </ul>
  )
}
