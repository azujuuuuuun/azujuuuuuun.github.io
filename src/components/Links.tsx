import React from 'react';
import { Link } from 'gatsby';

interface Props {
  listStyle?: object;
  isFooter?: boolean;
  itemStyle?: object;
  linkStyle?: object;
}

export default (props: Props) => {
  const { listStyle, isFooter, itemStyle, linkStyle } = props;
  return (
    <ul style={listStyle}>
      {isFooter && (
        <li style={itemStyle}>
          <Link to="/" style={linkStyle}>
            ホーム
          </Link>
        </li>
      )}
      <li style={itemStyle}>
        <Link to="/about" style={linkStyle}>
          自己紹介
        </Link>
      </li>
      <li style={itemStyle}>
        <Link to="/skills" style={linkStyle}>
          スキル
        </Link>
      </li>
      <li style={itemStyle}>
        <Link to="/favorites" style={linkStyle}>
          好きなもの
        </Link>
      </li>
      <li style={itemStyle}>
        <Link to="/links" style={linkStyle}>
          リンク
        </Link>
      </li>
    </ul>
  );
};
