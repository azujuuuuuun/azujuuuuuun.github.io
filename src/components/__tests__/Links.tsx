import React from 'react'
import renderer from 'react-test-renderer'

import Links from '../Links'

describe('Links', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Links />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
