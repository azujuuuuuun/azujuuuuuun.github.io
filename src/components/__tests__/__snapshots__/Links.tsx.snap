// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links renders correctly 1`] = `
<ul>
  <li>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      自己紹介
    </a>
  </li>
  <li>
    <a
      href="/favorites"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      好きなもの
    </a>
  </li>
  <li>
    <a
      href="/links"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      リンク
    </a>
  </li>
</ul>
`;

exports[`Links renders isFooter prop correctly 1`] = `
<ul>
  <li>
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      ホーム
    </a>
  </li>
  <li>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      自己紹介
    </a>
  </li>
  <li>
    <a
      href="/favorites"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      好きなもの
    </a>
  </li>
  <li>
    <a
      href="/links"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      リンク
    </a>
  </li>
</ul>
`;
